pkg/apis/core/types.go

```diff
@@ -2274,6 +2274,8 @@ type NodeSelector struct {
 type NodeSelectorTerm struct {
 	//Required. A list of node selector requirements. The requirements are ANDed.
 	MatchExpressions []NodeSelectorRequirement
+	//Optional. A list of node selector requirements by node's field. The requirements are ANDed.
+	MatchFields []NodeSelectorRequirement
```

Comment:
does this change MatchExpressions to be optional (at least one of MatchExpressions and MatchFields must be specified?), or would you expect this to be paired with a MatchExpression that selected everything (is that even possible)?

Reply:
> at least one of MatchExpressions and MatchFields must be specified

This one :) And the requirements are ANDed; if both are not empty, they also are ANDed.

Reply:
Done; updated comments and added validation. Does not check the key of `MatchFields`, will handle it when adding related logic in scheduler.